corehq/blobs/s3db.py

```diff
@@ -143,23 +140,41 @@ def _s3_bucket(self, create=False):
         return self.db.Bucket(self.s3_bucket_name)
 
 
-class BlobStream(ClosingContextProxy):
+class BlobStream(RawIOBase):
 
     def __init__(self, stream, blob_db, blob_key):
-        super(BlobStream, self).__init__(stream)
+        self._obj = stream
         self._blob_db = weakref.ref(blob_db)
         self.blob_key = blob_key
 
+    def readable(self):
+        return True
+
+    def read(self, *args, **kw):
+        return self._obj.read(*args, **kw)
+
+    read1 = read
+
+    def write(self, *args, **kw):
+        raise IOError
+
+    def tell(self):
+        return self._obj._amount_read
+
     def seek(self, offset, from_what=os.SEEK_SET):
```

Comment:
what does seek do here, if it is not supported? you can use it as noop and nothing else?

Reply:
It was added in https://github.com/dimagi/commcare-hq/pull/22719 because of [this line](https://github.com/dimagi/commcare-hq/blob/76b72b53f9ce1733a341f7c13f29f0d4bb6b2cd2/corehq/blobs/s3db.py#L75). The error in that PR got triggered when migrating blobs from one S3 backend to another. Having that `content.seek(0)` is messy for sure, and it would be nice if it could be removed.